File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> memo;
        for(int i = 0; i < nums.size(); i++) {
            if(memo.find(nums[i]) == memo.end()) 
                memo[target-nums[i]] = i;
            else return {memo[nums[i]], i};
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut dict = HashMap::new();
        for (index, num) in nums.iter().enumerate() {
            if let Some(&val) = dict.get(num) {
                return vec![val, index as i32];
            }
            dict.insert(target - num, index as i32);
        }
        Vec::new()
    }
}